### PR TITLE
feat(esbuild): new plugin for ESBuild & Stylable

### PR DESCRIPTION
**Pull request by barak007**
### ESBuild stylable plugin.

Currently works similar to the `experimental-loader` without depth calculations.
This is initial implementation to allow experimentations.

### CODE REVIEW

File: packages/esbuild/LICENSE [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Copyright (c) 2017, Wix.com Ltd.
```
Comment (AviVahl) [inline packages/esbuild/LICENSE:1]: 2022

File: packages/esbuild/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# @stylable/webpack-plugin
```
Comment (AviVahl) [inline packages/esbuild/README.md:1]: I've inserted the name below

File: packages/esbuild/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+[![npm version](https://img.shields.io/npm/v/@stylable/esbuild.svg)](https://www.npmjs.com/package/@stylable/esbuild)
+
+`@stylable/esbuild` 
+
+## License
+Copyright (c) 2017 Wix.com Ltd. All Rights Reserved. Use of this source code is governed by an [BSD license](./LICENSE).
```
Comment (AviVahl) [inline packages/esbuild/README.md:8]: MIT?

File: packages/esbuild/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  ],
+  "engines": {
+    "node": ">=12"
+  },
+  "publishConfig": {
+    "access": "public"
+  },
+  "repository": "https://github.com/wix/stylable/tree/master/packages/esbuild",
+  "author": "Wix.com",
+  "license": "BSD-3-Clause"
```
Comment (AviVahl) [inline packages/esbuild/package.json:33]: MIT

File: packages/esbuild/src/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "../../../tsconfig.base.json",
+  "compilerOptions": {
+    "outDir": "../dist"
+  },
+  "references": [{ "path": "../../core/src" }, { "path": "../../optimizer/src" }]
```
Comment (AviVahl) [inline packages/esbuild/src/tsconfig.json:6]: It also uses nod e

File: packages/esbuild/test/e2e/simple-project.spec.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+class ESBuildTestKit {
+    buildFile: string;
+    buildResult!: BuildResult;
+    constructor(public project: string) {
+        this.buildFile = require.resolve(`@stylable/esbuild/test/e2e/${this.project}/build`);
+    }
+    async build() {
+        const { run }: ProjectBuild = await import(this.buildFile);
+        this.buildResult = await run(esbuild, (options: BuildOptions) => ({
```
Comment (AviVahl) [inline packages/esbuild/test/e2e/simple-project.spec.ts:22]: Return res in promise. Why state?

File: packages/esbuild/test/e2e/simple-project/build.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+const { stylablePlugin } = require('@stylable/esbuild');
+
+module.exports.run = function run(build, o) {
+    return build(
+        o({
```
Comment (AviVahl) [inline packages/esbuild/test/e2e/simple-project/build.js:5]: o?

File: typings/externals/index.d.ts [hunk 8]
```diff
@@ -37,6 +37,13 @@ declare module 'node-eval' {
     export = nodeEval;
 }
 
+declare module '*.st.css' {
```
Comment (AviVahl) [inline typings/externals/index.d.ts:40]: Why?

File: packages/esbuild/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+                resolveDir: '.',
+                contents: `
+                import { sts, stc } from "@stylable/esbuild/runtime"; 
+                import ${JSON.stringify(args.path)};
+                export const namespace = ${JSON.stringify(res.meta.namespace)};
+                export const classes = ${JSON.stringify(res.exports.classes)};
+                export const keyframes = ${JSON.stringify(res.exports.keyframes)};
+                export const stVars = ${JSON.stringify(res.exports.stVars)};
+                export const vars = ${JSON.stringify(res.exports.vars)};
+                export const cssStates = /* @__PURE__ */ stc.bind(namespace);
```
Comment (goloveychuk) [inline packages/esbuild/src/index.ts:59]: `sts.bind(undefined, namespace)`

**Comment by barak007** [comment]
moved to https://github.com/wix/stylable/pull/2856

**Pull request closed without merging**